### PR TITLE
server: remove local rocksdb dsn.

### LINKED ISSUE #1206: remove local RocksDB DSN for tikv-server
We support a local RocksDB DSN for tikv-server, but 

+ We have never used it for running server. 
+ TiDB doesn't support it. 
+ The default DSN is `rocksdb`, user may use this wrongly. Although we must provide more detailed doc, I still think the best way is to prevent users from making this error.

### PR DESCRIPTION
**Pull request by disksing**
Fix #1206 

PTAL @siddontang @BusyJay 

Note:
After this PR, we should update tikv startup script by removing `-S raftkv`. But you don't have to make changes if use a config file before. @iamxy @nolouch @shenli

### CODE REVIEW

**Comment by siddontang** [comment]
Removing `-S raftkv` is not backward-compatible, we should keep it. 
We can add a `deprecated` tag for the flag first.

**Comment by siddontang** [comment]
Rest LGTM

**Comment by ngaut** [comment]
LGTM

File: src/bin/tikv-server.rs [hunk 1]
```diff
@@ -877,7 +854,7 @@ fn main() {
                 "default: 0 (unlimited)");
     opts.optopt("S",
                 "dsn",
-                "set which dsn to use, warning: default is rocksdb without persistent",
+                "[deprecated] set which dsn to use, warning: now only support raftkv",
                 "dsn: rocksdb, raftkv");
```
Comment (siddontang) [inline src/bin/tikv-server.rs]: remove rocksdb

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**